### ISSUE TITLE
UnboundLocalError: local variable 'delta' referenced before assignment

### ISSUE THREAD

**Issue by adamzenith** [continuedev/continue#554]
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: ubuntu 22.04
- Continue: 0.0.412 & v0.1.26
- Ollama: 0.1.2
```

### Description

Some variable delta is used before assignment in the code. 

### To reproduce

Select a large amount( for me 300 lines was enough) of lines of code using ctrl+m, then try and write a prompt.

### Log output

```Shell
Traceback (most recent call last):

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/core/autopilot.py", line 438, in _run_singular_step
    observation = await step(self.continue_sdk)

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/core/main.py", line 376, in __call__
    return await self.run(sdk)

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/plugins/steps/chat.py", line 105, in run
    async for chunk in generator:

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/libs/llm/base.py", line 418, in stream_chat
    messages = self.compile_chat_messages(

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/libs/llm/base.py", line 262, in compile_chat_messages
    return compile_chat_messages(

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/libs/util/count_tokens.py", line 214, in compile_chat_messages

  File "/home/matthias/.local/lib/python3.10/site-packages/continuedev/libs/util/count_tokens.py", line 115, in prune_chat_history
    while tokens_removed < distance_from_third[i] and total_tokens - total_tokens_removed > context_length:

UnboundLocalError: local variable 'delta' referenced before assignment
```

**Comment by sestinj** [comment]
@adamzenith thanks for sharing this - shipping a version tomorrow that will fix this!

**Comment by adamzenith** [comment]
Awesome!

**Comment by sestinj** [comment]
@adamzenith this should now be solved in pre-release version 0.1.28

**Comment by sestinj** [comment]
The solution to this issue is now included in the main release
